TypeCobol.Test/Report/ReportTestHelper.cs

```diff
@@ -73,8 +73,8 @@ public static ReturnCode ParseWithNodeListenerReportCompare(string fileName, str
                 parser.Init(input, typeCobolOption, format, new List<string>() { copyFolder });
                 parser.Parse(input);
 
-                var allDiags = parser.Results.AllDiagnostics();
-                if (allDiags.Count == 0)
+                // warning diagnostics are not considered : for example, test with warning with COPY SUPPRESS is always running
+                if (parser.Results.AllDiagnostics().Any(d => d.Info.Severity != TypeCobol.Compiler.Diagnostics.Severity.Warning) == false)
```

Comment:
I'd rather invert condition to avoid testing against false value :
```csharp
if (parser.Results.AllDiagnostics().Any(d => d.Info.Severity != TypeCobol.Compiler.Diagnostics.Severity.Warning))
{
    return ReturnCode.ParserDiagnosticsErrors;
}
else
{
    ...
}
```
But it's ok.

Reply:
Fix with replacing
if (parser.Results.AllDiagnostics().Any(d => d.Info.Severity != TypeCobol.Compiler.Diagnostics.Severity.Warning) == false)
with
if (parser.Results.AllDiagnostics().All(d => d.Info.Severity == TypeCobol.Compiler.Diagnostics.Severity.Warning))
https://github.com/TypeCobolTeam/TypeCobol/tree/774bbecec9a370b3d43ebfac4c3d9ab38cc7e072

Reply:
Actually there is also an Info severity (not used yet) so you can test if All diagnostics are different from error (to keep current code structure) or test if Any diagnostic is error (to reduce nesting).